internal/endpoints/utils.go

```diff
@@ -112,9 +116,73 @@ func validTimestamps(q structs.Query, all bool) bool {
 	return true
 }
 
+// Check for a specified role in the user's identity header, returns (200, nil) if the role is found
+func checkForRole(r *http.Request, role string) (int, error) {
+	identityHeader := r.Header.Get("x-rh-identity")
+	if identityHeader == "" {
+		return http.StatusUnauthorized, errors.New("Missing Identity Header")
+	}
+
+	type IdentityHeader struct {
+		Identity struct {
+			Associate struct {
+				Roles []string `json:"Role"`
+			} `json:"associate"`
+		} `json:"identity"`
+	}
+
+	var identityHeaderData IdentityHeader
+	// base64 decode the header
+	decoded, err := base64.StdEncoding.DecodeString(identityHeader)
+	if err != nil {
+		l.Log.Error("Error decoding identity header", "error", err)
+		return http.StatusUnauthorized, err
+	}
+
+	err = json.Unmarshal(decoded, &identityHeaderData)
+	if err != nil {
+		l.Log.Error("Error unmarshalling identity header", "error", err)
+		return http.StatusUnauthorized, err
+
+	}
+
+	if !stringInSlice(role, identityHeaderData.Identity.Associate.Roles) {
+		return http.StatusForbidden, errors.New("You do not have the required permissions to access this resource")
+	}
+
+	return http.StatusOK, nil
+}
+
 // Write HTTP Response
 func writeResponse(w http.ResponseWriter, status int, message string) {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(status)
 	w.Write([]byte(message))
 }
+
+// Send a request for an ArchiveLink to storage-broker
+func requestArchiveLink(r *http.Request, reqID string) (*structs.PayloadArchiveLink, error) {
+	req, err := http.NewRequest("GET", config.Get().StorageBrokerURL+"?request_id="+reqID, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is for a later phase. We should be fine without these for now, I think.

* Might be good idea to have a timeout when we send out this request, so that we don't wait for storage broker for too long.

* Should we have some rate limits here, so that we don't ddos this? (this scenario isn't very likely, but it might be good to that protection)